### PR TITLE
fix(pininput): on safari one-time-code input there is no event.data

### PR DESCRIPTION
**Pull request by Ryguy11**
## Describe the problem this PR addresses
In safari the one-time-code autofill option was not correctly filling out all the inputs in the pin-input. Instead the autocomplete code was being filled completely in the first input.
square-pay-code

## Describe the changes in this PR
The issue was the event.data of the autofill did not contain any data on Safari. Not exactly sure why to be honest, but we are able to work around this by reaching into `event.target.value` for the input.

### CODE REVIEW

**Review by tatems — APPROVED** [review]

File: src/components/PinInput/src/PinInput.vue [hunk 1]
```diff
@@ -126,13 +126,13 @@ export default {
 		},
 
 		onInputPin(event, index) {
-			if (!event.data) {
+			const input = event?.data || event?.target?.value;
```
Comment (landondurnan) [inline src/components/PinInput/src/PinInput.vue:129]: Could you add some comments here about the data values and the reason for the conditional? Also any external references regarding why safari works a bit differently if they exist?

Comment (pretzelhammer) [inline src/components/PinInput/src/PinInput.vue:129]: Can you help me understand why `event.data` was used in the first place? I'm use to seeing `event.target.value` everywhere, which seems to be the reliable cross-browser standard for accessing an input's value, so if you're gonna be using that you can probably just get rid of the `event.data`.

Comment (pretzelhammer) [reply src/components/PinInput/src/PinInput.vue:129]: Oh, does it have something to do with people trying to paste the entire code into the first input cell? 🤔

Comment (Ryguy11) [reply src/components/PinInput/src/PinInput.vue:129]: https://developer.mozilla.org/en-US/docs/Web/API/InputEvent

Youre right that it makes sense to just get rid of the event.data in this case

**Review by mattmewton — APPROVED** [review]

File: src/components/PinInput/src/PinInput.vue (hunk 1 above)
Comment (Ryguy11) [reply src/components/PinInput/src/PinInput.vue]: No - its just that the input event has a `.data` attribute that seems to have cross-browser support which is why its originally what I was using. On Safari it doesnt seem to be there for some reason, even though the docs say it supports  it. I also tried using `input.native` to see if for some reason that was messing with anything but it had the same effect

Comment (Ryguy11) [reply src/components/PinInput/src/PinInput.vue]: Im just going to remove the conditional as per Kirills suggestion

Comment (Ryguy11) [reply src/components/PinInput/src/PinInput.vue]: re: pasting the entire code into the first input cell - the component already supports that functionality 👍

**Review by pretzelhammer — APPROVED** [review]

File: src/components/PinInput/src/PinInput.vue (hunk 1 above)
Comment (pretzelhammer) [reply src/components/PinInput/src/PinInput.vue]: great, thanks for the context! i've resolved the open threads and approved the pr

**Pull request merged**